src/Yoti.Auth/Activity.cs

```diff
@@ -36,23 +36,57 @@ public ActivityDetails HandleSuccessfulResponse(AsymmetricCipherKeyPair keyPair,
             {
                 return new ActivityDetails(ActivityOutcome.SharingFailure);
             }
-            else
-            {
-                ReceiptDO receipt = parsedResponse.Receipt;
 
-                AttrpubapiV1.AttributeList attributes = CryptoEngine.DecryptCurrentUserReceipt(
-                    parsedResponse.Receipt.wrapped_receipt_key,
-                    parsedResponse.Receipt.other_party_profile_content,
-                    keyPair);
+            ReceiptDO receipt = parsedResponse.Receipt;
 
-                _yotiUserProfile.Id = parsedResponse.Receipt.remember_me_id;
-                _yotiProfile.Id = parsedResponse.Receipt.remember_me_id;
-                string receiptId = parsedResponse.Receipt.receipt_id;
+            ParseOtherPartyProfileContent(keyPair, parsedResponse);
+            Dictionary<string, BaseAttribute> applicationProfileAttributes = ParseApplicationProfileContent(keyPair, parsedResponse);
 
-                AddAttributesToProfile(attributes);
+            ApplicationProfile applicationProfile = new ApplicationProfile(applicationProfileAttributes);
+
+            DateTime? timestamp = null;
+            if (receipt.timestamp != null)
+            {
+                if (DateTime.TryParseExact(receipt.timestamp, "yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime parsedDate))
+                {
+                    timestamp = parsedDate;
+                }
+            }
 
-                return new ActivityDetails(_yotiUserProfile, _yotiProfile, receiptId, ActivityOutcome.Success);
+            return new ActivityDetails(parsedResponse.Receipt.remember_me_id, timestamp, _yotiUserProfile, _yotiProfile, applicationProfile, parsedResponse.Receipt.receipt_id, ActivityOutcome.Success);
```

Comment:
This is just a suggestion: Will it be better if pass the whole Receipt Object to ActivityDetails constructor?
As ActivityDetails(Receipt $receipt, ..) as rememberMe_Id, timestamp, and Receipt_id are all from Receipt.
The decision is yours, this is just a suggestion.

Reply:
That's a good suggestion, but since `ReceiptDO` has an `internal` access modifier and is less accessible than `ActivityDetails`, if we were to pass in the whole receipt data object, we'd have to convert `ReceiptDO` to `public`. I think we should keep it as internal, even though it would make it slightly neater if we passed the whole receipt in.